### PR TITLE
feat: cache evm addresses

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

⚠️ depends on https://github.com/iotaledger/firefly/pull/6880

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/components/popups/AddProposalPopup.svelte [hunk 1]
```diff
@@ -40,7 +40,7 @@
                 nodeInput?.validate(),
             ])
             await registerParticipationWrapper()
-            updateActiveAccountMetadata($selectedAccount.index, {
+            updateSelectedAccountPersistedData($selectedAccount.index, {
```
Comment (MarkNerdi) [inline packages/desktop/components/popups/AddProposalPopup.svelte:43]: this function should update the selectedAccount, therefor the index is not needed

Comment (MarkNerdi) [reply packages/desktop/components/popups/AddProposalPopup.svelte:43]: or we name it `updateAccountPersistedData` and update the logic in that function (see comment there)

File: packages/shared/lib/core/account/actions/updateSelectedAccountPersistedData.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { IPersistedAccountData, selectedAccountIndex, updateSelectedAccount } fr
 import { updateAccountPersistedDataOnActiveProfile } from '@core/profile'
 import { get } from 'svelte/store'
 
-export function updateActiveAccountPersistedData(
+export function updateSelectedAccountPersistedData(
```
[the remark this answers is not in the document]
Comment (begonaalvarezd) [reply packages/shared/lib/core/account/actions/updateSelectedAccountPersistedData.ts]: reverted this change 🙏🏼

File: packages/desktop/components/popups/AddProposalPopup.svelte (hunk 1 above)
Comment (begonaalvarezd) [reply packages/desktop/components/popups/AddProposalPopup.svelte]: reverted this change 🙏🏼

**Pull request merged**